languages/java/oso/src/main/java/com/osohq/oso/Host.java

```diff
@@ -167,15 +167,10 @@ public boolean isSubclass(String leftTag, String rightTag) {
     return rightClass.isAssignableFrom(leftClass);
   }
 
-  /** Check if two instances unify. */
-  public boolean unify(long leftId, long rightId) throws Exceptions.UnregisteredInstanceError {
-    Object left = getInstance(leftId);
-    Object right = getInstance(rightId);
-    if (left == null) {
-      return right == null;
-    } else {
-      return left.equals(right);
-    }
+  public boolean operator(String op, List<Object> initargs) throws Exceptions.OsoException {
+    Object l = initargs.get(0), r = initargs.get(1);
+    if (op.equals("Eq")) return l == null ? l == r : l.equals(r);
```

Comment:
This would be easier to read if it was less compressed, like the previous implementation of `unify`